Userland/Libraries/LibArchive/Tar.cpp

```diff
@@ -35,4 +35,21 @@ bool TarFileHeader::content_is_like_extended_header() const
     return type_flag() == TarFileType::ExtendedHeader || type_flag() == TarFileType::GlobalExtendedHeader;
 }
 
+void TarFileHeader::set_filename_and_prefix(StringView filename)
+{
+    if (filename.length() <= sizeof(m_filename)) {
+        set_prefix(""sv);
+        set_filename(filename);
+        return;
+    }
+
+    Optional<size_t> slash = filename.find('/');
+    for (slash = filename.find('/');
+         slash.has_value() && filename.length() - slash.value() > sizeof(m_filename);
+         slash = filename.find('/', slash.value() + 1)) { }
+
+    VERIFY(slash.has_value());
+    set_prefix(filename.substring_view(0, slash.value() + 1));
+    set_filename(filename.substring_view(slash.value() + 1));
```

Comment:
Can prefixes really only be split off on slashes? I don't recall seeing or having to rely on that when initially implementing support for prefixes, and if it isn't required it would drastically simplify this function.

Reply:
I initially implemented it that way, but then found that our unpacking did require it be split on a slash. I checked and GNU tar will not split filenames. if you give it a 120 character filename in ustar mode it will fail with "filename too long"

Reply:
I found a reference for this: https://pubs.opengroup.org/onlinepubs/7908799/xcu/pax.html

> A new pathname is formed, if prefix is not an empty string (its first character is not NUL), by concatenating prefix (up to the first NUL character), a slash character and name ; otherwise, name is used alone.

note that pax is posix's tar for some reason.

Reply:
Ah, don't mind me then. I also just noticed that what I used back then is a path prepend, not a string prepend.